plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -457,11 +483,24 @@ private static boolean isIdentityColumn(JsonNode node)
                 .anyMatch(name -> name.startsWith("delta.identity."));
     }
 
-    public static Map<String, String> getColumnInvariants(MetadataEntry metadataEntry)
+    public static Map<String, String> getColumnInvariants(MetadataEntry metadataEntry, ProtocolEntry protocolEntry)
     {
+        if (protocolEntry.supportsWriterFeature()) {
+            if (!protocolEntry.writerFeaturesContains("invariants")) {
+                return ImmutableMap.of();
```

Comment:
Why we check for checkConstraints now and we didn't before?
If the previous logic was flawed, and this is a fix, it would be nice to have it as a separate commit.